File: new/has_duplicates_hash_set.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        // create an unordered_set for tracking seen items
        // if seen return true, else return false
        unordered_set<int> seen;
        for (int num: nums) {
            if (seen.count(num)) {
                // seen.count(num) returns either 0 or 1 always
                // because of unordered_set
                return true;
            }

            seen.insert(num);
        }

        return false;
        // time complexity : O(n)
        // space complexity : O(n)
    }
};

File: new/python/has_duplicates_hash_set.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        # create a set and iteratively add elements if it is not
        # already there, if they are present return True
        # finally return false
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False
        # Time complexity : O(n)
        # Space complexity : O(n)
